bond_app/datastore_cache_api.py

```diff
@@ -35,6 +35,9 @@ def get(self, key, namespace=None):
             return None
         return entry.value
 
+    def delete(self, key, namespace=None):
+        DatastoreCacheApi._build_cache_key(key, namespace).delete()
```

Comment:
Maybe the answer is that we just leave a comment here with a link to the docs and maybe copy/paste some of the text here, just in case google modifies it or takes it away.  We can just leave a note to our future selves saying that Google's documentation was not clear about actually gets "deleted" when we call delete.  Some sort of breadcrumb to our future selves that if this delete seems to not be fully deleting things.

Reply:
I think it just means you call `.delete()` on the key object, not the entity object. I don't think we need to worry about entities without keys. According to the Datastore documentation, all entities have a unique key https://cloud.google.com/datastore/docs/concepts/entities. 

As for it always returning `None`, I think it just means whether or not it finds the key you're trying to delete. I'd think that any exceptions would still be raised, but I can look into it.

Reply:
I've tested it with a datastore emulator and confirmed that the changes in this PR do indeed clear out the cached SA key from datastore